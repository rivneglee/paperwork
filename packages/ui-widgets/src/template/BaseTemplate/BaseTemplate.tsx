import React, { FunctionComponent, ReactElement } from 'react';
import { Scrollable } from '../../layout/Scrollable';

interface Props {
  header?: ReactElement;
  footer?: ReactElement;
  setScrollerRef?: (ref: any) => void;
  spinner?: ReactElement;
  isProcessing?: boolean;
}

const BaseTemplate: FunctionComponent<Props> = ({
  header,
  footer,
  children,
  setScrollerRef,
  spinner,
  isProcessing,
}) => {
  const templateView = (
    <div className="pw-template">
      {header && (
        <div className="pw-template__header">
          {header}
        </div>
      )}
      <div className="pw-template__scrollwrapper">
        <Scrollable className="pw-template__scroller" setRef={setScrollerRef}>
          <div className="pw-template__content">
            <div className="pw-template__inner">
              {children}
            </div>
          </div>
        </Scrollable>
      </div>
      {footer && (
        <div className="pw-template__footer">
          {footer}
        </div>
      )}
    </div>
  );

  return spinner && isProcessing ? spinner : templateView;
};

export default BaseTemplate;
