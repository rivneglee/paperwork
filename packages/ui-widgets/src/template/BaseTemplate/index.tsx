export { default as BaseTemplate } from './BaseTemplate';
