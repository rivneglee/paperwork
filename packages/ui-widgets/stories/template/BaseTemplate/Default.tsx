import React from 'react';

import { BaseTemplate, List, Icons } from '../../../src';

export default () => (
  <BaseTemplate>
    <List>
      {
        Array.from({ length: 10 }).map(() => (
          <>
            <List.Item onClick={close} icon={<Icons.Form/>}>
              Forms
            </List.Item>
            <List.Item onClick={close} icon={<Icons.Chart/>}>
              Reports
            </List.Item>
          </>
        ))
      }
    </List>
  </BaseTemplate>
);
