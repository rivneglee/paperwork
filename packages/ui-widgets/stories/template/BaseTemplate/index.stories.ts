import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Base template', module)
  .add('default', Default);
