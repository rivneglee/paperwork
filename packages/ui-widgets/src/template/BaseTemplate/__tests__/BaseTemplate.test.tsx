import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BaseTemplate from '../BaseTemplate';

describe('BaseTemplate', () => {
  const Header = () => <>Header</>;
  const Content = () => <>Content</>;
  const Footer = () => <>Footer</>;

  it('should render properly', () => {
    const wrapper = shallow(
      <BaseTemplate header={<Header/>} footer={<Footer/>}>
        <Content/>
      </BaseTemplate>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
