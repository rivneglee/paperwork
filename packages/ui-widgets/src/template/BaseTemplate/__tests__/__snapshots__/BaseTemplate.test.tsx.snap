// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTemplate should render properly 1`] = `
<div
  className="pw-template"
>
  <div
    className="pw-template__header"
  >
    <Header />
  </div>
  <div
    className="pw-template__scrollwrapper"
  >
    <Scrollable>
      <div
        className="pw-template__content"
      >
        <div
          className="pw-template__inner"
        >
          <Content />
        </div>
      </div>
    </Scrollable>
  </div>
  <div
    className="pw-template__footer"
  >
    <Footer />
  </div>
</div>
`;
